src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -582,6 +629,8 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
                 r = slash_boot_in_fstab();
                 if (r < 0)
                         return r;
+                if (r > 0 && fstab_is_same_device("/boot", p->node))
+                        return 0;
```

Comment:
You already explained in the commit message. But could you also add a short comment here?